l10n_fr_payment_payfip/models/payment_transaction.py

```diff
@@ -0,0 +1,159 @@
+            idop = res.acquirer_id.payfip_get_id_op_from_web_service(
+                email, amount, reference, acquirer_reference)
+            res.payfip_operation_identifier = idop
+
+        return res
+
+    def _get_specific_rendering_values(self, processing_values):
+        res = super()._get_specific_rendering_values(processing_values)
+        if self.provider != 'payfip':
+            return res
+
+        base_url = self.acquirer_id.get_base_url()
+        return {
+            'api_url': PayFIPController._payment_url,
+            'numcli': self.acquirer_id.payfip_customer_number,
+            'exer': fields.Datetime.now().year,
+            'refdet': self.acquirer_reference,
+            'objet': self.reference,
+            'montant': self.amount,
+            'mel': self.partner_email,
+            'urlnotif': urls.url_join(base_url, PayFIPController._notification_url),
+            'urlredirect': urls.url_join(base_url, PayFIPController._return_url),
+            'saisie': 'W' if self.acquirer_id.state == 'prod' else 'T',
+        }
+
+    @api.model
+    def _get_tx_from_feedback_data(self, provider, data):
+        """ Override of payment to find the transaction based on Payfip data.
+
+        param str provider: The provider of the acquirer that handled the transaction
+        :param dict data: The feedback data sent by the provider
+        :return: The transaction if found
+        :rtype: recordset of `payment.transaction`
+        :raise: ValidationError if the data match no transaction
+        """
+        tx = super()._get_tx_from_feedback_data(provider, data)
+        if provider != 'payfip':
+            return tx
+
+        reference = data["objet"]
+        tx = self.sudo().search(
+            [('reference', '=', reference), ('provider', '=', 'payfip')])
+        if not tx:
+            raise ValidationError(
+                "PayFIP: " +
+                _("No transaction found matching reference %s.", reference)
+            )
+        return tx
+
+    def _process_feedback_data(self, feedback_data):
+        data = self.acquirer_id.payfip_get_result_from_web_service(
+            feedback_data.get('idOp'))
+
+        refdet = feedback_data.get('refdet', False)
+        self.acquirer_reference = refdet
+
+        if data.get('code'):
+            self._set_pending()
+
+        self.ensure_one()
+
+        result = data.get('resultrans', False)
+        if not result:
+            self._set_pending()
+
+        payfip_amount = int(data.get('montant', 0)) / 100
+        if result in ['P', 'V']:
+            self._set_done()
+            self.write({
+                'payfip_state': result,
+                'payfip_amount': payfip_amount,
+            })
+            return True
+        elif result in ['A']:
+            message = 'Received notification for PayFIP payment %s: set as canceled' % self.reference
+            _logger.info(message)
+            self._set_canceled()
```

Comment:
I think you could translate error message and pass that message to [_set_canceled method](https://github.com/odoo/odoo/blob/15.0/addons/payment/models/payment_transaction.py#L687)

```suggestion
        elif result in ['A']:
            message = _('Received notification for PayFIP payment %s: set as canceled') % self.reference
            _logger.info(message)
            self._set_canceled(message)
```